### DISCUSSION TITLE
Clarification of the use of the AGPL along with the "ee" package

### DISCUSSION THREAD

**Discussion by ssddanbrown** [documenso/documenso#1415] [General]
Hello :wave: ,
To be up-front, I'm interested in licensing scenarios & techniques within the open source world.
I saw the project was under the AGPLv3 license, but I also saw the [ee folder and license](https://github.com/documenso/documenso/blob/1a9dcadba5876e89fe2e684a1ccf1675fae40103/packages/ee/LICENSE), so assumed an open-core & dual-license setup.

Looking further though, It looks like a lot of the open source code [relies on the ee package](https://github.com/search?q=repo%3Adocumenso%2Fdocumenso+%22%40documenso%2Fee%22&type=code&p=1), or at least has direct references to it unless there are other mechanisms at play, which from my own (non-legal-expert) experience of the AGPL I know does not mix too well.

So my questions are:
- Can this be built & ran on the open (AGPLv3 in the context of this project) code alone without any further required modifications?
  - _(Can the `packages/ee` folder be removed completely without causing build/run problems)_
- Since the project (at least the [community edition](https://docs.documenso.com/users/licenses/community-edition)) is distributed under the AGPLv3, what would stop a user from following [section 7 of the license](https://github.com/documenso/documenso/blob/1a9dcadba5876e89fe2e684a1ccf1675fae40103/LICENSE#L376-L380) to just remove any incompatible "further restrictions" added by your ee license? _(and therefore effectively adopt the ee code under the AGPL too)_

If I've misunderstood, and you actually have some open mocks/versions of that `ee` package, it would be great to know how that's done, so I can use that as a good example of mixed-license setups to show others.

Edit: Following the [self-hosted guidance](https://github.com/documenso/documenso?tab=readme-ov-file#self-hosting), a clone, `npm install` and a `npm run build:web` works fine by default, but attempting to run `npm run build:web` after removing the `packages/ee` folder seems to break the build with many `Module not found: Can't resolve '@documenso/ee/server-only/limits/provider/client'` kind of errors.

**Comment by ElTimuro** [comment]
@ssddanbrown thanks for the great write up. The details of our licensing are still evoling, the philosophy behind doesn't:

- Your characterization is correct, combining the the AGPL community edition and the EE licensed folder makes us open core
- You currently can't techncially build the project without the folder. As the enterprise segment is never in our history, it didn't seem neccessary to allow this yet. Doing so would make sense, this is a good hint
- The short answer to the Section 7 is, the EE part is not licensed under AGPLV3, so section 7 doesn't apply
- From our side it's fine if your build it with the EE folder, just using the features without an enterprise license would be illegal

- Legal technicalities aside, the goal of the setup is to have a clear legal requirement for enterprises hosting Documenso, needing to buy a license
- Another goal is to have a powerful, free community edition, with as little restrictions as possible
- Defining a few feature (that are not relevant for most) as enteprise seemed the obvious way to do this
- As the current focus of Documenso the company is to earn money to reeinvest into the OSS project, we wil keep this model to build a compelling enterprise offering, that does not rely on the goodwill of enterpriseses to pay. Hence the restriction.
- This isn't perfect, since I personally wouldn't mind the average Joe using enterprise features. It does however make it harder to sell if it's techically also free.
- For now, we resolve this by this dual licensing
- I wouldn't mind finetuning this in the future to allow even enterprise usage by the community edition, it has to be balanced with a sellable product though.

@Mythie making the build possible without EE is a good idea

**Reply by ssddanbrown** [reply]
Thanks for the response @ElTimuro!

> The short answer to the Section 7 is, the EE part is not licensed under AGPLV3, so section 7 doesn't apply

Ultimately this would probably need the view of legal experts but I think there's a strong case that the code is currently provided as part of the program, and that you're providing the program under the AGPLv3.  Just look at the repo page as you land on it. The readme shows plain AGPLv3. The LICENSE shows plain AGPLv3. Then [this page](https://docs.documenso.com/users/licenses/community-edition) specifically states that "Documenso and the Community Edition are licensed under [AGPL3](https://github.com/documenso/documenso/blob/main/LICENSE).". It can certainly given the impression that project as a whole (for which this non-open code is required for) are provided under the AGPLv3 which section 7 will then apply for.

> From our side it's fine if your build it with the EE folder, just using the features without an enterprise license would be illegal

Sure, but the the issue is that users will be technically relying-upon/using code of this commercial license without knowing it.

> [...] the goal of the setup is to have a clear legal requirement for enterprises hosting Documenso, needing to buy a license [...] It does however make it harder to sell if it's techically also free. [...] it has to be balanced with a sellable product though. [...]

Open source is about giving up rights to others, which does come with its own real challenges. Some of your goals seem in contradiction with the fundamental definition of open source (open use, modification & distribution) which will lead to ongoing friction as you try achieve both.

> I wouldn't mind finetuning this in the future to allow even enterprise usage by the community edition [...]

The AGPLv3 allows enterprise usage. Under the common definitions of free and open source in general you can't license against certain use-cases or audiences. You will continue to be in contention with open source and your license if that's your wish.

---

Just to be clear, I have no issues with licensing your efforts as needed or desired, and I have no issues with source-available or dual-licensing in general, I just think that transparency and clarity to users is quite important in these areas.

**Comment by ElTimuro** [comment]
@ssddanbrown happy to keep discussing this, as it's something I think a lot about
